### PR TITLE
feat: interaction validity endpoint

### PR DESCRIPTION
**Pull request by asiaziola**

### CODE REVIEW

File: src/common.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const { getLastStateFromDreCache, insertState } = require('./db/nodeDb');
+const warp = require('./warp');
+
+module.exports.isTxIdValid = (txId) => {
```
Comment (ppedziwiatr) [inline src/common.js:4]: 🙏

File: src/routes/validity.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  if (!isTxIdValid(contractId)) {
+    ctx.throw(400, 'Contract id not valid.');
+  }
+
+  try {
+    const { result, parsed } = await getContractState(contractId, nodeDb);
+
+    if (result) {
+      const validity = parsed ? result.validity : JSON.parse(result.validity);
```
Comment (ppedziwiatr) [inline src/routes/validity.js:26]: please add some simple caching here (e.g. Map-based LRU with ~1000 capacity?)

Comment (asiaziola) [reply src/routes/validity.js]: done 🍭

File: src/routes/validity.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  if (!isTxIdValid(id)) {
+    ctx.throw(400, 'Interaction id not valid.');
+  }
+
+  if (!isTxIdValid(contractId)) {
+    ctx.throw(400, 'Contract id not valid.');
+  }
+
+  try {
+    const contractValidity = validityCache.get(contractId);
```
Comment (ppedziwiatr) [inline src/routes/validity.js:26]: hmm..tbh I though about storing a map from interaction_id -> result, not contract_id -> full_validity

if we're gonna store full contract validity, reduce the capacity to sth like 20...as full contract validity and error messages can be BIG.

Comment (ppedziwiatr) [reply src/routes/validity.js:26]: ok, let's leave the current version, but please reduce the capacity to 20.

Comment (asiaziola) [reply src/routes/validity.js:26]: ok, done

**Pull request merged**